metaflow/plugins/kfp/kfp.py

```diff
@@ -398,15 +402,21 @@ def create_kfp_pipeline_from_flow_graph(self) -> Callable:
         step_to_kfp_component_map = self.create_kfp_components_from_graph()
 
         # Container op that corresponds to a step defined in the Metaflow flowgraph.
-        step_op = kfp.components.func_to_container_op(
-            step_op_func, base_image=self.base_image
+        step_op_component: Dict = yaml.load(
+            kfp.components.func_to_component_text(
+                _step_op_func, base_image=self.base_image
+            ),
+            yaml.SafeLoader,
         )
 
+        def step_op(name: str) -> Callable[..., ContainerOp]:
+            my_dict = dict(step_op_component)
+            my_dict["name"] = name
+            return kfp.components.load_component_from_text(yaml.dump(my_dict))
```

Comment:
Hmm maybe I'm missing the nuance here but isn't the name already set on line 434? Is this a different name?

Reply:
`set_display_name` is only metadata for the KFP UI to render in the UI.  Removing it because it's redundant (as you point out), and we can set the KFP component name instead, which is rendered in the UI.